barretenberg/cpp/src/barretenberg/commitment_schemes/zeromorph/zeromorph.hpp

```diff
@@ -321,14 +321,14 @@ template <typename Curve> class ZeroMorphProver_ {
                       auto&& g_shift_evaluations,
                       auto& multilinear_challenge,
                       auto& commitment_key,
-                      auto& transcript,
+                      auto transcript,
```

Comment:
The entire code could use a scrub for these sorts of things but doesn't it still make sense to have the ref here?

Reply:
Wait I take this back. I think we virtually always want to pass a shared ptr by value otherwise it defeats the purpose of the smart pointer..

Reply:
Adam looked at best practices and they say to pass by `const&` by default :shrug:

Reply:
@ludamad Tying to understand the thinking here. If we want to avoid the ref copy virtually everywhere, why not use a unique pointer? Is the point that we sometimes may want to genuinely use the shared pointer functionality? Otherwise it seems a bit like using a vector instead of an array for something whose size isn't allowed to change.

Reply:
But a shared pointer that is alive for the whole duration of a function may as well be a raw pointer?

Reply:
Really dereferencing a shared pointer is only bad if you store the dereferenced result in a structure that actually wants the ref count, there's no reason to ref count something that will live the entire duration of a function anyway